src/libcore/str/mod.rs

```diff
@@ -347,6 +371,18 @@ impl<'a> DoubleEndedIterator for CharIndices<'a> {
 #[derive(Clone)]
 pub struct Bytes<'a>(Map<slice::Iter<'a, u8>, BytesDeref>);
 
+impl<'a> Bytes<'a> {
+    /// View the underlying data as a subslice of the original data.
+    ///
+    /// This has the same lifetime as the original slice, and so the
+    /// iterator can continue to be used while this exists.
+    #[unstable(feature = "iter_to_slice", issue = "27775")]
+    #[inline]
+    pub fn as_str(&self) -> &'a str {
```

Comment:
Can this safely be done? We don't know that iteration isn't in the middle of a multi byte character, right?